File: android/measure-android/measure/src/main/java/sh/measure/android/attributes/PatchIdAttributeProcessor.kt
package sh.measure.android.attributes

import sh.measure.android.attributes.Attribute.PATCH_ID_KEY
import sh.measure.android.attributes.Attribute.PATCH_VERSION_KEY

/**
 * Maintains the state for the patch_id and patch_version attributes. Both are set once at SDK
 * initialization and identify the current OTA patch for sourcemap symbolication.
 */
internal class PatchIdAttributeProcessor : AttributeProcessor {
    private var patchId: String? = null
    private var patchVersion: String? = null

    override fun appendAttributes(attributes: MutableMap<String, Any?>) {
        attributes[PATCH_ID_KEY] = patchId
        attributes[PATCH_VERSION_KEY] = patchVersion
    }

    fun setPatchId(patchId: String) {
        this.patchId = patchId
    }

    fun setPatchVersion(patchVersion: String) {
        this.patchVersion = patchVersion
    }
}

File: ios/Sources/MeasureSDK/Swift/Attribute/PatchIdAttributeProcessor.swift
import Foundation

/// Maintains the state for the patch_id and patch_version attributes. Both are set once at SDK
/// initialization and identify the current OTA patch for sourcemap symbolication.
final class PatchIdAttributeProcessor: AttributeProcessor {
    private var patchId: String?
    private var patchVersion: String?

    func appendAttributes(_ attributes: Attributes) {
        attributes.patchId = patchId
        attributes.patchVersion = patchVersion
    }

    func setPatchId(_ patchId: String) {
        self.patchId = patchId
    }

    func setPatchVersion(_ patchVersion: String) {
        self.patchVersion = patchVersion
    }
}
